ci/aws/aws-insts.tf

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019 Cable Television Laboratories, Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# AWS EC2 Instances
+resource "aws_spot_instance_request" "snaps-boot-host" {
+  ami = var.ami
+  spot_type = "one-time"
```

Comment:
Can we make this configurable? We should have a workflow for both CI where waiting is good and one for develpment where we can't

Reply:
Are you referring to wait_for_fulfillment? Yes I can make this configurable.

Reply:
I'm cool with using a spot as long as I'm generally getting one in less than a minute or so.
I wasn't sure if 'spot_type' was part of it as well. Also am perplexed as to why flipping that flag would cause the script to crash and burn. If that is the case, I guess hardcoding is ok...

Reply:
@spisarski If there are no spot instances available when you request one, it will either wait for one or if it doesn't wait it will just fail.

It doesn't look like TF has an easy fallback option to failback to an expensive instance if a spot instance isn't available within a timeout.  That could be added as higher-level logic, but I would add that as another issue.

For development, if a spot instance isn't available fast enough, then we would want to configure the instance request type instead of the wait.

You can always check your e-mail while you wait ;)